lib/chef/resource/macos_userdefaults.rb

```diff
@@ -73,34 +107,38 @@ def coerce_booleans(val)
       end
 
       load_current_value do |desired|
-        value = coerce_booleans(desired.value)
-        cmd = "defaults read '#{desired.domain}' "
-        cmd << "'#{desired.key}' " if desired.key
-        cmd << " | grep -qx '#{value}'"
-
-        vc = if desired.user.nil?
-               shell_out(cmd)
-             else
-               shell_out(cmd, user: desired.user)
-             end
-
-        is_set !vc.error?
+        coerced_value = coerce_booleans(desired.value)
+
+        state_cmd = ['/usr/bin/defaults', 'read', desired.domain, desired.key]
+
+        state = if desired.user.nil?
+          shell_out(state_cmd)
+        else
+          shell_out(cmd, user: desired.user)
+        end
+
+        current_value_does_not_exist! if state.error?
```

Comment:
This should probably dump the error to the debug log or something in case of an unexpected error.

Reply:
Added debug logging